src/components/ComparisonTable.tsx

```diff
@@ -0,0 +1,166 @@
+  from: string;
+  to: string;
+  count: number;
+}
+
+type Tmaintainers={
+  username:string;
+   email:string;
+}
+
+type Trows={
+  packageName:string, firstPackageValue: string|number, secondPackageValue: string|number
+}
+
+
+export default function ComparisonTable({ selectedPackages }) {
+  const [firstPackagedata, setFirstPackageData] = useState<Tpackage| undefined>();
+  const [secondPackageData, setSecondPackageData] = useState<Tpackage| undefined>();
+  const [rows, setRows] = useState<Array<Trows>>([]);
+
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const response1 = await axios.get<Tpackage>(`https://api.npms.io/v2/package/${selectedPackages[0].package.name}`);
+        setFirstPackageData(response1.data);
+      } catch (error) {
+        console.error(error);
+      }
+    };
+    fetchData();
+  }, [selectedPackages]);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const response2 = await axios.get<Tpackage>(`https://api.npms.io/v2/package/${selectedPackages[1].package.name}`);
+        setSecondPackageData(response2.data);
+      } catch (error) {
+        console.error(error);
+      }
+    };
+    fetchData();
+  }, [selectedPackages]);
+
+  useEffect(() => {
+    if (firstPackagedata&&secondPackageData) {
+      setRows([
+        createData('Description', firstPackagedata.collected.metadata.name, secondPackageData.collected.metadata.description),
+        createData('Keywords', firstPackagedata.collected.metadata.keywords?firstPackagedata.collected.metadata.keywords[0]:"N/A", secondPackageData.collected.metadata.keywords?secondPackageData.collected.metadata.keywords[0]:"N/A"),
+        createData('License', firstPackagedata.collected.metadata.license?firstPackagedata.collected.metadata.license:"N/A",secondPackageData.collected.metadata.license?secondPackageData.collected.metadata.license:"N/A"),
+        createData('Last Modification Date', firstPackagedata.analyzedAt.slice(0,10), secondPackageData.analyzedAt.slice(0,10)),
+        createData('Authors/Publishers', firstPackagedata.collected.metadata.author?firstPackagedata.collected.metadata.author.name:"N/A", secondPackageData.collected.metadata.author?secondPackageData.collected.metadata.author.name:"N/A"),
+        createData('Maintainers', firstPackagedata.collected.metadata.maintainers[0]?firstPackagedata.collected.metadata.maintainers[0].username:"null", secondPackageData.collected.metadata.maintainers[0]?secondPackageData.collected.metadata.maintainers[0].username:"null"),
+      ]);
+    }
+  }, [firstPackagedata, secondPackageData]);
+
+  function createData(
+    packageName: string,
+    firstPackageValue:string|number,
+    secondPackageValue:string|number,
+  ):Trows {
+    return { packageName, firstPackageValue, secondPackageValue };
+  }
+
+  return (
+    <>
+      {firstPackagedata &&secondPackageData&& (
+        <>
+       <TableContainer component={Paper} elevation={5} sx={{
```

Comment:
Create a separate component for Table and pass only the rowData from here as props.